### PR TITLE
Fixed CAInfoService.getKRAInfoClient()

### PR DESCRIPTION
**Pull request by edewata**
The `CAInfoService.getKRAInfoClient()` and
`CAService.getConnector()` have been modified to use the
client certificate specified in the CA's KRA connector to
access KRA. If the client certificate is missing, it will
use the subsystem certificate instead.

The `CAInfoService` has also been modified to propagate
any exception during the above operation to the caller.

https://bugzilla.redhat.com/show_bug.cgi?id=1861911

### CODE REVIEW

File: base/ca/src/com/netscape/ca/CAService.java [hunk 1]
```diff
@@ -288,9 +285,13 @@ public IConnector getConnector(IConfigStore config)
             RemoteAuthority remauthority =
                     new RemoteAuthority(host, port, uri, timeout);
 
-            // Change end
-            if (nickname == null)
+            // Use client cert specified in KRA connector (i.e. subsystem cert)
+            String nickname = config.getString("nickName", null);
+            if (nickname == null) {
+                // Use CA signing cert as client certificate
                 nickname = mCA.getNickname();
```
Comment (ladycfu) [inline base/ca/src/com/netscape/ca/CAService.java:292]: I know this came from old code, but  I don't think we should attempt to use ca signing cert as client cert.

File: base/ca/src/org/dogtagpki/server/rest/CAInfoService.java [hunk 2]
```diff
@@ -180,11 +183,23 @@ private static KRAInfoClient getKRAInfoClient(KRAConnectorInfo connInfo)
...
+
         ClientConfig config = new ClientConfig();
         int port = Integer.parseInt(connInfo.getPort());
         config.setServerURL("https", connInfo.getHost(), port);
         config.setNSSDatabase(cs.getInstanceDir() + "/alias");
 
+        // Use client cert specified in KRA connector (i.e. subsystem cert)
+        String nickname = kraConnectorConfig.getString("nickName", null);
+        if (nickname == null) {
+            // Use CA signing cert as client certificate
```
Comment (ladycfu) [inline base/ca/src/org/dogtagpki/server/rest/CAInfoService.java:198]: same here.  Let it fail instead.

File: base/ca/src/com/netscape/ca/CAService.java (hunk 1 above)
Comment (edewata) [reply base/ca/src/com/netscape/ca/CAService.java:292]: Agreed. I'd suggest to simplify the code as follows:
```
String nickname = config.getString("nickName");
```
The code will fail if `nickName` parameter is missing, so we're guaranteed to have a nickname.

File: base/ca/src/org/dogtagpki/server/rest/CAInfoService.java (hunk 2 above)
Comment (edewata) [reply base/ca/src/org/dogtagpki/server/rest/CAInfoService.java:198]: Agreed. Is the suggested change above OK?

File: base/ca/src/com/netscape/ca/CAService.java (hunk 1 above)
Comment (ladycfu) [reply base/ca/src/com/netscape/ca/CAService.java:292]: I dont' have a full view of the code, but if the exception is properly handled then yes.

**Comment by edewata** [comment]
Please see the updated PR. Thanks!

**Comment by edewata** [comment]
Thanks!

**Pull request merged**